lib/prompts/downloadProjectPrompt.ts

```diff
@@ -1,36 +1,55 @@
-// @ts-nocheck
-const { promptUser } = require('./promptUtils');
-const { getAccountId } = require('@hubspot/local-dev-lib/config');
-const { fetchProjects } = require('@hubspot/local-dev-lib/api/projects');
-const { logError, ApiErrorContext } = require('../errorHandlers/index');
-const { EXIT_CODES } = require('../enums/exitCodes');
-const { i18n } = require('../lang');
+import { promptUser } from './promptUtils';
+import { getAccountId } from '@hubspot/local-dev-lib/config';
+import { fetchProjects } from '@hubspot/local-dev-lib/api/projects';
+import { logError, ApiErrorContext } from '../errorHandlers/index';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { EXIT_CODES } from '../enums/exitCodes';
+import { i18n } from '../lang';
+import { Project } from '@hubspot/local-dev-lib/types/Project';
 
 const i18nKey = 'lib.prompts.downloadProjectPrompt';
 
-const createProjectsList = async accountId => {
+type DownloadProjectPromptResponse = {
+  project: string;
+};
+
+async function createProjectsList(
+  accountId: number | null
+): Promise<Project[]> {
   try {
-    const { data: projects } = await fetchProjects(accountId);
-    return projects.results;
+    if (accountId) {
+      const { data: projects } = await fetchProjects(accountId);
+      return projects.results;
+    }
+    logger.error(i18n(`${i18nKey}.errors.accountIdRequired`));
+    process.exit(EXIT_CODES.ERROR);
   } catch (e) {
-    logError(e, new ApiErrorContext({ accountId }));
+    if (accountId) {
+      logError(e, new ApiErrorContext({ accountId }));
+    } else {
+      logError(e);
+    }
```

Comment:
Nit: this could be `logError(e, accountId ? new ApiErrorContext({ accountId }) : undefined)`